pkg/query-service/app/integrations/traceFunnels/funnel.go

```diff
@@ -0,0 +1,145 @@
+package traceFunnels
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+type FunnelStore struct {
+	sync.RWMutex
+	funnels map[string]*Funnel
+}
+
+func (s *FunnelStore) CreateFunnel(name, userID, orgID string, timestamp int64) (*Funnel, error) {
+	s.Lock()
+	defer s.Unlock()
+
+	for _, existingFunnel := range s.funnels {
+		if existingFunnel.Name == name && existingFunnel.CreatedBy == userID {
+			return nil, fmt.Errorf("funnel with name '%s' already exists for user '%s'", name, userID)
+		}
+	}
+
+	if timestamp == 0 {
+		return nil, fmt.Errorf("timestamp is required")
+	}
+
+	funnel := &Funnel{
+		ID:        uuid.New().String(),
+		Name:      name,
+		CreatedAt: timestamp * 1000000, // Convert milliseconds to nanoseconds for internal storage
+		CreatedBy: userID,
+		OrgID:     orgID,
+		Steps:     make([]FunnelStep, 0),
+	}
+
+	s.funnels[funnel.ID] = funnel
```

Comment:
why is this getting stored in memory.

Reply:
I wanted to fall back to store in memory if connection to db fails